src/OpenTelemetry.Collector.Dependencies/DependenciesCollector.cs

```diff
@@ -33,18 +35,25 @@ public class DependenciesCollector : IDisposable
         /// Initializes a new instance of the <see cref="DependenciesCollector"/> class.
         /// </summary>
         /// <param name="options">Configuration options for dependencies collector.</param>
-        /// <param name="tracer">Tracer to record traced with.</param>
-        /// <param name="sampler">Sampler to use to sample dependnecy calls.</param>
-        public DependenciesCollector(DependenciesCollectorOptions options, ITracer tracer, ISampler sampler)
+        /// <param name="tracerFactory">TracerFactory to create a Tracer to record traced with.</param>
+        /// <param name="sampler">Sampler to use to sample dependency calls.</param>
+        public DependenciesCollector(DependenciesCollectorOptions options, ITracerFactory tracerFactory, ISampler sampler)
         {
+            var tracer = tracerFactory.GetTracer(typeof(DependenciesCollector).Namespace, typeof(DependenciesCollector).Assembly.GetName().Version.FormatResourceVersion());
             this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber<HttpRequestMessage>(
-                new Dictionary<string, Func<ITracer, Func<HttpRequestMessage, ISampler>, ListenerHandler<HttpRequestMessage>>>()
+                new Dictionary<string, Func<ITracerFactory, Func<HttpRequestMessage, ISampler>, ListenerHandler<HttpRequestMessage>>>()
                 {
-                    { "HttpHandlerDiagnosticListener", (t, s) => new HttpHandlerDiagnosticListener(t, s) },
-                    { "Azure.Clients", (t, s) => new AzureSdkDiagnosticListener("Azure.Clients", t, sampler) },
-                    { "Azure.Pipeline", (t, s) => new AzureSdkDiagnosticListener("Azure.Pipeline", t, sampler) },
+                    {
```

Comment:
so named tracers are intended to help with filtering. correct? Then we probably want to filter `HttpHandlerDiagnosticListener`, `Azure.Clients`,  `Azure.Pipeline` separately - they are signals coming from different libraries, i.e. they should get separate tracers based on source name. Now they get the same tracers and you can only turn all of them on or nothing.

Also, with .NET for these signals we have a different filtering way (through DiagnosticSource subscription) which is way more efficient than named tracers as it prevents instrumentation inside the library from doing anything. 

We should address it in another work item https://github.com/open-telemetry/opentelemetry-dotnet/issues/251

Reply:
Basically yes: Named Tracers allow to associate produced spans with a specific Tracer (or instrumentation library) that produced them.
This further allows for disabling Tracers or having other options specific for those.

That's why I was not 100% sure how this case fits into the picture. The actual data signals come from different libraries here than were the Tracer is instantiated.

I'm happy to implement your suggestion though.